### PR TITLE
[mobile] Add HTTP/3 related API in starting_envoy instruction

### PR DESCRIPTION
**Pull request by RenjieTang**
Commit Message: Add HTTP/3 related API in starting_envoy instruction
Additional Description: enableHttp3() has long been available. addQuicHint() was recently added in https://github.com/envoyproxy/envoy/pull/29224
Risk Level: Low
Testing: n/a
Docs Changes: Envoy Mobile api doc change
Release Notes: n/a
Platform Specific Features: mobile only

### CODE REVIEW

File: mobile/docs/root/api/starting_envoy.rst [hunk 1]
```diff
@@ -363,6 +363,36 @@ Specify whether to enable transparent response Brotli decompression. Defaults to
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``enableHttp3``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Specify whether to enable HTTP/3. Defaults to false. Only available when the Envoy Mobile build has HTTP/3 included.
+
+**Example**::
+
+  // Kotlin
+  builder.enableHttp3(true)
```
Comment (alyssawilk) [inline mobile/docs/root/api/starting_envoy.rst:375]: Do we have this function in Kotlin? I don't see it in library/kotlin/io/envoyproxy/envoymobile/EngineBuilder.kt

File: mobile/docs/root/api/starting_envoy.rst [hunk 2]
```diff
@@ -363,6 +363,36 @@ Specify whether to enable transparent response Brotli decompression. Defaults to
...
+  // Swift
+  builder.enableHttp3(true)
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``addQuicHint``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Add a host port pair that's known to support QUIC. Only available when HTTP/3 is enabled.
+It can be called multiple times to append a list of QUIC hints.
+This allows HTTP/3 to be used for the first request to the hosts, thus saving a TCP round trip.
```
Comment (alyssawilk) [inline mobile/docs/root/api/starting_envoy.rst:386]: If we do the above we can "see enableHttp3 for details"

**Review by alyssawilk — COMMENTED** [review]
Thanks for following up on this!
/wait

File: mobile/docs/root/api/starting_envoy.rst (hunk 1 above)
Comment (RenjieTang) [reply mobile/docs/root/api/starting_envoy.rst:377]: Surprised we don't have it in Kotlin. Added the API.

File: mobile/docs/root/api/starting_envoy.rst [hunk 3]
```diff
@@ -363,6 +363,36 @@ Specify whether to enable transparent response Brotli decompression. Defaults to
 Default values from the `brotli decompressor proto <https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/compression/brotli/decompressor/v3/brotli.proto>`_
 are used.
 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``enableHttp3``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Specify whether to enable HTTP/3. Defaults to false. Only available when the Envoy Mobile build has HTTP/3 included.
```
[the remark this answers is not in the document]
Comment (RenjieTang) [reply mobile/docs/root/api/starting_envoy.rst]: done

File: mobile/docs/root/api/starting_envoy.rst (hunk 2 above)
Comment (RenjieTang) [reply mobile/docs/root/api/starting_envoy.rst]: done.

**Review by alyssawilk — APPROVED** [review]

**Comment by alyssawilk** [comment]
is CI wedged?  maybe a main merge would help?

**Pull request merged**